src/ClockPanel.tsx

```diff
@@ -66,6 +66,7 @@ export function ClockPanel(props: Props) {
         now={now}
         replaceVariables={props.replaceVariables}
         options={props.options}
+        data={data}
```

Comment:
you are passing all the data and responsibility of calculate your field to a component that should merely render data. Instead, create another function that abstracts all the logic and returns the field you actually want and then pass it here.

Reply:
you also want a loading state. since you can only use data when it is done. You will want to show a [spinner](https://developers.grafana.com/ui/latest/index.html?path=/story/visualizations-spinner--basic) while the data loads when the source is a query or it'll show blank while it loads.

Reply:
I dont know if you want to show a spinner every time the data loads. Grafana itself shows if a query is done and this logic isn't taking that much more time.

Reply:
you are right! Let's just leave it as is :)